bluesky/callbacks/fitting.py

```diff
@@ -273,16 +273,16 @@ def compute(self):
             _x = x[cr:cr+2]
             _y = y[cr:cr+2] - mid
 
-            dx = np.diff(_x)
-            dy = np.diff(_y)
+            dx = np.diff(_x)[0]
+            dy = np.diff(_y)[0]
             m = dy / dx
             _cen_list.append((-_y[0] / m) + _x[0])
 
         if _cen_list:
             self.cen = np.mean(_cen_list)
 
-        if len(_cen_list) == 2:
-            self.fwhm = float(_cen_list[1] - _cen_list[0])
+        if len(_cen_list) >= 2:
+            self.fwhm = np.abs(_cen_list[-1] - _cen_list[0], dtype=float)
```

Comment:
Thanks @danielballan  I haven't used this. However, looking at this, I have a general set of comments:

1. one probably should document `fwhm` in line 201.

2. personally, I'd argue that it doesn't make sense (and might confuse the user) when more than two crossings are found. But it's probably better to output more data than nothing. I would add more to the comment so the user can know that this has been computed on data that contained more than two crossings. How about including like @mrakitin suggested a `fwhm_extended` and `fwhm`. Maybe `fwhm` and `fwhm2`?
Or at least something like `ncrossings`, which is the number of crossings found. But that may be overkill.
Another thing I noticed, if `cen_list < 2` we should probably set `self.fwhm = None` since this is a re-usable object (and same for `cen`).


3. In passing, I also found something that may be a little awkward. Why not use:
```py
def center_of_mass(y, x=None):
if x is None:
    x = np.arange(len(y))
return np.sum(x*y)/np.sum(y)
```
instead? There may be a reasoning I'm not seeing. (maybe set `y = np.abs(y)` as well)


I think for this PR, just documenting the fwhm, and setting `fwhm` and `cen` to `None` where appropriate is probably enough. Suggested documentation:
```
fwhm : the computed full width half maximum (fwhm) of a peak. 
This is done by computing the number of times the data crosses the median.
The distance between the first and last crossing is taken to be the fwhm.
```

You could maybe add something like crossings (just suggestion)
```
crossings : the points at which the data crosses the median
```
just for completeness. (i.e. what does my `fwhm` mean? for one peak, or for multiple crossings?